### DISCUSSION TITLE
Is there a command separator?

### DISCUSSION THREAD

**Discussion by AmirSimantov** [plantuml/plantuml#1419] [General]
Hi,

For documentation purposes, I would like to write an entire script in a single line, instead of multiple lines.
I there some command separator that I can use such as semi-colon in C-Style languages? Or maybe a replacement for a new line?

For example, instead of:

@startuml
(*) --> "Create a sample data spreadsheet"
--> "Look for data of a new pattern"

I would like to write:
@startuml;(*) --> "Create a sample data spreadsheet";--> "Look for data of a new pattern"

Thanks!

**Comment by The-Lum** [comment]
Hi @AmirSimantov, and all,

>Or maybe a replacement for a new line?

You can just use `\n` like:
```puml
@startuml\n(*) --> "Create a sample data spreadsheet"\n--> "Look for data of a new pattern"
```
_If that can help,_
Regards.

**Comment by AmirSimantov** [comment]
Thanks but it doesn't work... I get the instructions of how to build the script.

**Comment by VladimirAlexiev** [comment]
@AmirSimantov Newlines are significant in PlantUML, and many diagrams include `\n` to put a newline in the middle of a label.
So you'll need to encode the diagram source (eg as JSON string),a nd decode it before processing.

Though why you want to have the diagram in one row, is beyond my understanding.
BTW, did you know that the plantuml server encodes the diagram text in a URL string? See https://plantuml.com/text-encoding
